.github/workflows/build_and_test.yml

```diff
@@ -9,6 +9,9 @@ jobs:
       - uses: actions/setup-python@v4
         with:
           python-version: "3.8.x"
+      - name: Install dependencies
+        run: |
+          sudo apt install cppcheck
```

Comment:
This pre-commit hook requires an external installation of `cppcheck`. I tried to run it on a Windows machine without installing it and it give a generic error:
```shell
FileNotFoundError: [WinError 2] The system cannot find the file specified
```

This might be a pretty bad user experience for new contributors...

Regardless, we don't have any documentation about our pre-commit hooks - probably the best place to put it is in [`CONTRIBUTING.md`](https://github.com/seladb/PcapPlusPlus/blob/master/CONTRIBUTING.md), and there we can mention the requirement of installing `cppcheck`. What do you think?

Reply:
I updated `CONTRIBUTING.md`. One option might be removing cppcheck from pre-commit and only checking at Github actions while encouraging people to use from command line, but it might be worse experience.